k-distribution/src/main/scripts/lib/k

```diff
@@ -1,4 +1,36 @@
 #!/usr/bin/env bash
 source "$(dirname "$0")/setenv"
-ulimit -s `ulimit -H -s`
-eval $JAVA org.kframework.main.Main '"$@"'
+ulimit -s "$(ulimit -H -s)"
+
+KERNEL_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/../../../../../../kernel" && pwd)"
+K_BASE="$(cd "$(dirname "${BASH_SOURCE[0]}")/../" && pwd)"
+SNAPSHOT_JAR="$(find "${KERNEL_DIR}"/target/ -name '*-SNAPSHOT.jar')"
+
+if [[ -z "${SNAPSHOT_JAR}" ]]; then
+    echo "Could not find the kernel jar file. Please run 'mvn package' in the K source directory." 1>&2
+    exit 1
+fi
+
+for flag in "$@"; do
+    if [[ "$flag" == "--version" || "$flag" == "-v" ]]; then
+        version=$(unzip -p "${SNAPSHOT_JAR}" META-INF/MANIFEST.MF | grep 'Implementation-Git-Describe' | awk -v RS='\r' '{print $2}')
+        versionTimestamp=$(unzip -q -c "${SNAPSHOT_JAR}"| grep "Implementation-Date:" | awk -v RS='\r' '{print $2}')
+        versionTimestampInMilliseconds=$(echo "${versionTimestamp} / 1000" | bc)
```

Comment:
Try to get the info directly from `get-version-string.sh`
This is the script begin called by mvn when packaging the K tools.